ADR/ADR013-use-paper-trail-gem-for-auditing-and-record-backups.md

```diff
@@ -0,0 +1,51 @@
+# ADR013: Add PaperTrail gem for auditing and record backup
+
+Date: 2023-02-27
+
+## Status
+
+Accepted
+
+## Context
+
+We need way to audit changes to database records to identify what has changed and who made those changes. This information
+will be surfaced to users later on when we come to versioning. This is still valuable to record now for any support tickets
+that come in and require us to understand the form/pages history.
+
+There is also a secondary need be able to restore individual records rather than having to restore from database backups.
```

Comment:
I haven't seen this need, do you have a link to a Trello or similar? Is this a requirement from versioning and offering the ability for admins to revert to an earlier version or as some kind of support activity by us? Links to my question below about how this would actually be done.

How does papertrail deal with versions for records that have been deleted? Does it keep their versions and can they be restored in the way you're intending?

Reply:
There isn't a specific Trello card for it. I am aware that there has been discussions around offering users the option to possibly rollback etc but its still very early days. I think for the time being this is kind a support activity/tool for us to be able to revert changes (if theres a large number of request we can look at either making it a super admin feature or perhaps allow specific users with the correct permissions to do it). What I was trying to do was point out that as part of capturing audit trails there is also a way to restore the records to their given state.

To answer you delete question, papertrail logs `deletes` as an event and as part of that loging it stores the state that the record was in when it was deleted. This means that we can restore the record by pulling out the delete event and using the state the record was last in to restore it. https://github.com/paper-trail-gem/paper_trail#3a-reverting-and-undeleting-a-model

Reply:
Great thanks for the clarification Al. I don't wish to be pedantic but I wonder if this should be phrased as "There is also a secondary benefit where it could be possible to restore..." since "need" to me implies something concrete thats on the roadmap.

Reply:
No problem, suggestions more than help 😄 👍